tutor/hooks/consts.py

```diff
@@ -0,0 +1,266 @@
+"""
+List of all the action, filter and context names used across Tutor. This module is used
+to generate part of the reference documentation.
+"""
+# The Tutor plugin system is licensed under the terms of the Apache 2.0 license.
+__license__ = "Apache 2.0"
+
+from tutor import hooks
+
+__all__ = ["Actions", "Filters", "Contexts"]
+
+
+class Actions:
```

Comment:
If you take a look at [openedx-events](https://github.com/openedx/openedx-events/blob/main/openedx_events/learning/signals.py) (which seems like the Open edX app-level analogue to Tutor's actions), the event names are all adjectives, describing the state of the system in terms of something that has happened or is happening. For example, there is an event named `COURSE_ENROLLMENT_CREATED`, as opposed to `COURSE_ENROLL` or `COURSE_ENROLL_POST`. I like this because it reinforces what an event is: something happens in the core, and plugins can hook in some callbacks in order to respond to that thing.

Would you be willing to rename the actions under that framing? `CORE_READY` and `CORE_ROOT_READY` are great. For the others, I might suggest:
* `plugins:discover` -> `plugins:discovering`
* `plugin:load:{0}` -> `plugins:loading:{0}`
* `plugins:load-post` -> `plugins:loaded`

Reply:
Good point. I also like the fact that this allows us to rename `VERB_NOUN` to `NOUN_LOADED` (e.g: `LOAD_PLUGINS` -> `PLUGINS_LOADED`) which allows us to better sort actions by namespace.

However I'm not a big fan of the "-ing" form. This leads to me to realize that the "plugins:discover" action is not really necessary, as it's only triggered on `CORE_READY`. So I think that I'm going to get rid of it. With this in mind, I'll do the following changes:

- plugins:discover -> replace by core:ready
- plugin:load:{0} -> plugins:loaded:{0} I've really racked my brains for this one. "Loaded" does not really reflect the event that is occurring. It's more like "we are going to load the plugin because it is requested/required/necessary". But I like none of these adjectives, so I've decided to stick with "loaded". It's probably ok if we modify the action name later, once we find a better name, as it's only used internally.
- plugin:load -> get rid of it
- plugins:load-post -> plugins:loaded